tests/e2e/vm_service_utils.go

```diff
@@ -0,0 +1,586 @@
+		},
+		"type": "SUBSCRIBED"
+	}`, contentlibName, dsMoId, contentLibUrl, sslThumbPrint)
+
+	resp, statusCode := invokeRestAPI4PostRequests(vcRestSessionId, contentlbCreationUrl, reqBody)
+	gomega.Expect(statusCode).Should(gomega.BeNumerically("==", 201))
+
+	gomega.Expect(json.Unmarshal(resp, &contentLibId)).NotTo(gomega.HaveOccurred())
+	gomega.Expect(contentLibId).NotTo(gomega.BeEmpty())
+	framework.Logf("Successfully created content library 'csi-vmsvc-test-content-lib' for the test: %v", contentLibId)
+
+	return contentLibId
+}
+
+// getContentLibId4Url fetches ID of a content lib that matches the given URL
+func getContentLibId4Url(vcRestSessionId string, url string) (string, error) {
+	var libId string
+	libIds := getAllContentLibIds(vcRestSessionId)
+	for _, libId := range libIds {
+		lib := getContentLib(vcRestSessionId, libId)
+		if lib.url == url {
+			return libId, nil
+		}
+	}
+	return libId, fmt.Errorf("couldn't find a content lib with subscription url '%v'", url)
+}
+
+// getAllContentLibIds fetches IDs of all content libs
+func getAllContentLibIds(vcRestSessionId string) []string {
+	vcIp := e2eVSphere.Config.Global.VCenterHostname
+	contentLibsFetchUrl := "https://" + vcIp + "/api/content/subscribed-library"
+
+	resp := invokeRestAPI4GetRequests(vcRestSessionId, contentLibsFetchUrl)
+
+	v := []string{}
+	gomega.Expect(json.Unmarshal(resp, &v)).NotTo(gomega.HaveOccurred())
+	framework.Logf("Content lib IDs:\n%v", v)
+	return v
+}
+
+// getContentLib fetches the content lib with give ID
+func getContentLib(vcRestSessionId string, libId string) subscribedContentLibBasic {
+	vcIp := e2eVSphere.Config.Global.VCenterHostname
+	contentLibFetchUrl := "https://" + vcIp + "/api/content/subscribed-library/" + libId
+
+	resp := invokeRestAPI4GetRequests(vcRestSessionId, contentLibFetchUrl)
+
+	var v map[string]interface{}
+	gomega.Expect(json.Unmarshal(resp, &v)).NotTo(gomega.HaveOccurred())
+
+	var cl subscribedContentLibBasic
+	cl.name = v["name"].(string)
+	cl.Id = v["id"].(string)
+	cl.ds_moId = v["storage_backings"].([]interface{})[0].(map[string]interface{})["datastore_id"].(string)
+	cl.url = v["subscription_info"].(map[string]interface{})["subscription_url"].(string)
+
+	framework.Logf("Content lib with id %v: %v", libId, cl)
+	return cl
+}
+
+// invokeRestAPI4GetRequests invokes GET on given VC REST URL using the passed session token and verifies that the
+// return status code is 200
+func invokeRestAPI4GetRequests(vcRestSessionId string, url string) []byte {
+	transCfg := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	httpClient := &http.Client{Transport: transCfg}
+	framework.Logf("Invoking GET on url: %s", url)
+	req, err := http.NewRequest("GET", url, nil)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	req.Header.Add("vmware-api-session-Id", vcRestSessionId)
+
+	resp, statusCode := httpRequest(httpClient, req)
+	gomega.Expect(statusCode).Should(gomega.BeNumerically("==", 200))
```

Comment:
Can we return the status code as well? We might use this method for some negative usecase

Reply:
done